File: src/main/java/main/utils/resume/GuildResumeCache.java
package main.utils.resume;

import com.fasterxml.jackson.core.JsonProcessingException;
import lombok.extern.slf4j.Slf4j;
import main.utils.database.mongodb.cache.redis.RedisCache;

@Slf4j
public class GuildResumeCache extends RedisCache {

    public GuildResumeCache(String guildID) {
        super("resume:" + guildID);
    }

    public void setTracks(ResumeData resumeData) {
        set(resumeData.toString());
    }

    public ResumeData loadData() throws JsonProcessingException {
        final var tracks = ResumeData.fromJSON(get());
        if (del() == 0L)
            log.warn("Attempted to delete tracks to resume after load but nothing was deleted! Cache ID: {}", getCacheID());
        return tracks;
    }

    public boolean hasTracks() {
        return exists();
    }
}

File: src/main/kotlin/main/utils/resume/GuildResumeCache.kt
package main.utils.resume

import main.utils.database.mongodb.cache.redis.RedisCache
import org.slf4j.LoggerFactory

class GuildResumeCache(guildId: String) : RedisCache("resume:$guildId") {

    companion object {
        private val logger = LoggerFactory.getLogger(Companion::class.java)
    }

    var data: ResumeData?
        get() {
            val cachedData = get() ?: return null
            val tracks = ResumeData.fromJSON(cachedData)
            if (del() == 0L)
                logger.warn("Attempted to delete tracks to resume after load but nothing was deleted! Cache ID: $cacheID")
            return tracks
        }
        set(value) {
            set(value.toString())
        }

    val hasTracks: Boolean = exists()
}